internal/runtime/client/client.go

```diff
@@ -153,6 +163,98 @@ func (c *client) Unregister(extensionConfig *runtimev1.ExtensionConfig) error {
+func (c *client) CallAllExtensions(ctx context.Context, hook catalog.Hook, request runtime.Object, response runtimehooksv1.Response) error {
+	gvh, err := c.catalog.GroupVersionHook(hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registrations, err := c.registry.List(gvh.GroupHook())
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ExtensionHandlers for %s", gvh.GroupHook())
+	}
+	responses := []runtimehooksv1.Response{}
+	for _, registration := range registrations {
+		// Creates a new instance of the response parameter.
+		tmpResponse := reflect.New(reflect.TypeOf(response).Elem()).Interface().(runtimehooksv1.Response)
+		err = c.CallExtension(ctx, hook, registration.Name, request, tmpResponse)
+		// If one of the extension handlers fails lets short-circuit here and return early.
+		if err != nil {
+			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
+		}
+		responses = append(responses, response)
+	}
+	if err := aggregateResponses(responses); err != nil {
+		return errors.Wrap(err, "failed to aggregate responses")
+	}
+	return nil
+}
+
+func aggregateResponses(responses []runtimehooksv1.Response) error {
+	// TODO:(killianmuldoon) implement proper aggregation logic.
+	return nil
+}
+
+// CallExtension make the call to the extension with the given name.
+// The response object passed will be updated with the response of the call.
+// An error is returned if the extension is not compatible with the hook.
+// If the ExtensionHandler returns a response with `Status` set to `Failure` the function returns an error
+// and the response object is updated with the response received from the extension handler.
+//
+// FailurePolicy of the ExtensionHandler is used to handle errors that occur when performing the external call to the extension.
+// - If FailurePolicy is set to Ignore, the error is ignored and the response object is updated to be the default success response.
+// - If FailurePolicy is set to Fail, an error is returned and the response object may or may not be updated.
+// Nb. FailurePolicy does not affect the following kinds of errors:
+// - Internal errors. Examples: hooks is incompatible with ExtensionHandler, ExtensionHandler information is missing.
+// - Error when ExtensionHandler returns a response with `Status` set to `Failure`.
+func (c *client) CallExtension(ctx context.Context, hook catalog.Hook, name string, request runtime.Object, response runtimehooksv1.Response) error {
+	gvh, err := c.catalog.GroupVersionHook(hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registration, err := c.registry.Get(name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ExtensionHandler with name %q", name)
+	}
+	if gvh.GroupHook() != registration.GroupVersionHook.GroupHook() {
+		return fmt.Errorf("ExtensionHandler %q does not match group %s, hook %s", name, gvh.Group, gvh.Hook)
```

Comment:
nit: Let's use consistently `github.com/pkg/errors`
(note this applies to multiple places)